<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerMarkupsToModelModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerMarkupsToModelModuleWidget">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>443</width>
    <height>911</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Markups To Model</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="ParameterLayout">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>7</number>
     </property>
     <item row="0" column="1">
      <widget class="qMRMLNodeComboBox" name="ParameterNodeSelector">
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLMarkupsToModelNode</string>
        </stringlist>
       </property>
       <property name="renameEnabled">
        <bool>true</bool>
       </property>
       <property name="selectNodeUponCreation">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <layout class="QGridLayout" name="ModeSelectionLayout">
       <item row="0" column="0">
        <widget class="QRadioButton" name="ModeClosedSurfaceRadioButton">
         <property name="text">
          <string>Closed surface</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QRadioButton" name="ModeCurveRadioButton">
         <property name="text">
          <string>Curve</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="ParameterNodeLabel">
       <property name="text">
        <string>Parameter node:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="ModeLabel">
       <property name="text">
        <string>Mode:</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="MarkupsLayout">
     <property name="bottomMargin">
      <number>20</number>
     </property>
     <item>
      <widget class="qSlicerSimpleMarkupsWidget" name="MarkupsSelector"/>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QGridLayout" name="ModelNodeLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="ModelNodeLabel">
       <property name="text">
        <string>Model Output Node:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="qMRMLNodeComboBox" name="ModelNodeSelector">
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLModelNode</string>
        </stringlist>
       </property>
       <property name="noneEnabled">
        <bool>true</bool>
       </property>
       <property name="renameEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ctkMenuButton" name="UpdateButton">
     <property name="text">
      <string>Update</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleGroupBox" name="AdvancedGroupBox">
     <property name="title">
      <string>Advanced Model Options</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="collapsed">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="10" column="0">
       <widget class="QLabel" name="KochanekBiasLabel">
        <property name="text">
         <string>Kochanek Bias:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="ButterflySubdivisionLabel">
        <property name="text">
         <string>Butterfly Subdivision:</string>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QDoubleSpinBox" name="DelaunayAlphaDoubleSpinBox">
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="DelaunayAlphaLabel">
        <property name="text">
         <string>Delaunay Alpha:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="TubeRadiusLabel">
        <property name="text">
         <string>Tube Radius:</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="KochanekTensionLabel">
        <property name="text">
         <string>Kochanek Tension:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="qMRMLSpinBox" name="TubeFacesSpinBox">
        <property name="decimals">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <layout class="QHBoxLayout" name="SplineInterpolationLayout">
        <item>
         <widget class="QRadioButton" name="KochanekInterpolationRadioButton">
          <property name="text">
           <string>Kochanek</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="CardinalInterpolationRadioButton">
          <property name="text">
           <string>Cardinal</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="LinearInterpolationButton">
          <property name="text">
           <string>Linear</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="10" column="1">
       <widget class="QDoubleSpinBox" name="KochanekBiasDoubleSpinBox">
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="TubeSegmentsLabel">
        <property name="text">
         <string>Tube Segments Per Point:</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="SplineInterpolationLabel">
        <property name="text">
         <string>Spline Interpolation:</string>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QDoubleSpinBox" name="KochanekTensionDoubleSpinBox">
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="11" column="0">
       <widget class="QLabel" name="KochanekContinuityLabel">
        <property name="text">
         <string>Kochanek Continuity</string>
        </property>
       </widget>
      </item>
      <item row="11" column="1">
       <widget class="QDoubleSpinBox" name="KochanekContinuityDoubleSpinBox">
        <property name="minimum">
         <double>-1.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="TubeRadiusDoubleSpinBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="qMRMLSpinBox" name="TubeSegmentsSpinBox">
        <property name="decimals">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="TubeFacesLabel">
        <property name="text">
         <string>Tube Faces:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QCheckBox" name="ConvexHullResultCheckBox">
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="ButterflySubdivisionCheckBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="ConvexHullResultLabel">
        <property name="text">
         <string>Convex Hull on Result</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="CleanMarkupsLabel">
        <property name="text">
         <string>Clean Duplicated Markups:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="CleanMarkupsCheckBox">
        <property name="toolTip">
         <string extracomment="It will merge duplicate  points.  Duplicate points are the ones closer than tolerance distance. The tolerance distance is 1% diagonal size bounding box of total points."/>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleGroupBox" name="CollapsibleGroupBox">
     <property name="title">
      <string>Display Options</string>
     </property>
     <property name="collapsed">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="ModelVisibilityLabel">
        <property name="text">
         <string>Model Visibility:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="ModelOpacityLabel">
        <property name="text">
         <string>Model Opacity:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="ModelColorLabel">
        <property name="text">
         <string>Model Color:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkSliderWidget" name="ModelOpacitySlider">
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="pageStep">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="ModelVisiblityButton">
        <property name="toolTip">
         <string>Turns visibility on for all models (does not include hierarchies)</string>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>:/Icons/Small/SlicerVisible.png</normaloff>:/Icons/Small/SlicerVisible.png</iconset>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="ctkSliderWidget" name="MarkupsTextScaleSlider">
        <property name="maximum">
         <double>20.000000000000000</double>
        </property>
        <property name="value">
         <double>4.500000000000000</double>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="ModelSliceIntersectionCheckbox">
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="ModelSliceIntersectionsLabel">
        <property name="text">
         <string>Model Slice Intersections:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="ctkColorPickerButton" name="ModelColorSelector"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="MarkupsTextScaleLabel">
        <property name="text">
         <string>Markups Text Scale:</string>
        </property>
       </widget>
      </item>
     </layout>
     <zorder>ModelOpacityLabel</zorder>
     <zorder>ModelOpacitySlider</zorder>
     <zorder>ModelColorLabel</zorder>
     <zorder>MarkupsTextScaleLabel</zorder>
     <zorder>MarkupsTextScaleSlider</zorder>
     <zorder>ModelVisiblityButton</zorder>
     <zorder>ModelVisibilityLabel</zorder>
     <zorder>ModelColorSelector</zorder>
     <zorder>ModelSliceIntersectionCheckbox</zorder>
     <zorder>ModelSliceIntersectionsLabel</zorder>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="ActionUpdateAuto">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Auto-Update</string>
   </property>
  </action>
  <action name="ActionUpdateManual">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Update</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLSpinBox</class>
   <extends>ctkDoubleSpinBox</extends>
   <header>qMRMLSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerSimpleMarkupsWidget</class>
   <extends>qSlicerWidget</extends>
   <header>qSlicerSimpleMarkupsWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkColorPickerButton</class>
   <extends>QPushButton</extends>
   <header>ctkColorPickerButton.h</header>
  </customwidget>
  <customwidget>
   <class>ctkDoubleSpinBox</class>
   <extends>QWidget</extends>
   <header>ctkDoubleSpinBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkMenuButton</class>
   <extends>QPushButton</extends>
   <header>ctkMenuButton.h</header>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>qSlicerMarkupsToModelModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ParameterNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>414</x>
     <y>9</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerMarkupsToModelModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ModelNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerMarkupsToModelModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MarkupsSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>66</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>79</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
